storage/innobase/srv/srv0mon.cc

```diff
@@ -1687,14 +1687,14 @@ srv_mon_process_existing_counter(
 
 	/* innodb_row_lock_time_avg */
 	case MONITOR_OVLD_LOCK_AVG_WAIT_TIME:
-		mysql_mutex_lock(&lock_sys.wait_mutex);
+		lock_sys.wait_mutex_lock();
```

Comment:
I do not think that reads from `INFORMATION_SCHEMA.innodb_metrics` are time-critical enough to warrant any spinning here. This counter `lock_row_lock_time_avg` is enabled by default, though.